src/realm/util/interprocess_mutex.hpp

```diff
@@ -131,6 +132,11 @@ class InterprocessMutex {
     void free_lock_info();
 #else
     SharedPart* m_shared_part = nullptr;
+
+#ifdef _WIN32
+    HANDLE m_handle = 0;
```

Comment:
From the docs that I read, it appears that we should be using `INVALID_HANDLE_VALUE` instead of `0` because it is implementation dependent; `INVALID_HANDLE_VALUE` may be "the pseudohandle returned by GetCurrentProcess()" ([reference](https://blogs.msdn.microsoft.com/oldnewthing/20040302-00/?p=40443)). This being the case, we need to also be careful to check `if (m_handle != INVALID_HANDLE_VALUE)` everywhere instead of `if (m_handle)`. This will apply to other handles in the code as well including  `InterprocessCondvar::m_sema` and `InterprocessCondvar::m_waiters_done`.